File: src/main/java/de/hinundhergestellt/jhuh/vendors/ready2order/ArtooProductClient.java
package de.hinundhergestellt.jhuh.vendors.ready2order;

import de.hinundhergestellt.jhuh.vendors.ready2order.api.ProductApi;
import org.springframework.stereotype.Component;

import java.util.stream.Stream;

@Component
public class ArtooProductClient {

    private final ProductApi api;

    public ArtooProductClient(ApiClient apiClient) {
        api = new ProductApi(apiClient);
    }

    public Stream<ArtooProduct> findAll() {
        return PagingIterator.stream(this::findAll);
    }

    public ArtooProduct findById(int id) {
        return new ArtooProduct(api.productsIdGet(id, true, null, null));
    }

    public void save(ArtooProduct product) {
        product.save(api);
    }

    private Stream<ArtooProduct> findAll(int page) {
        return api.productsGet(page, null, null, null, null, null, null, true, null, null).stream()
                .map(ArtooProduct::new);
    }
}

File: src/main/java/de/hinundhergestellt/jhuh/vendors/ready2order/ArtooProductClient.kt
package de.hinundhergestellt.jhuh.vendors.ready2order

import de.hinundhergestellt.jhuh.vendors.ready2order.openapi.ApiClient
import de.hinundhergestellt.jhuh.vendors.ready2order.openapi.api.ProductApi
import org.springframework.stereotype.Component

@Component
class ArtooProductClient(
    apiClient: ApiClient
) {
    private val api = ProductApi(apiClient)

    fun findAll() = pageAll {
        api.productsGet(it, null, null, null, null, null, null, true, null, null)
            .map { product -> ArtooProduct(product) }
    }

    fun findById(id: Int): ArtooProduct {
        return ArtooProduct(api.productsIdGet(id, true, null, null))
    }

    fun save(product: ArtooProduct) {
        product.save(api)
    }

    fun delete(product: ArtooProduct) {
        api.productsIdDelete(product.id)
    }
}
